trie/trie.go

```diff
@@ -670,3 +670,129 @@ func (t *Trie) Reset() {
 	t.tracer.reset()
 	t.committed = false
 }
+
+// ReviveTrie revives a trie by prefix key with the given proof list.
+func (t *Trie) ReviveTrie(key []byte, prefixKeyHex []byte, proofList [][]byte) {
+	key = keybytesToHex(key)
+	newRoot, _, err := t.revive(t.root, key, prefixKeyHex, proofList, 0)
+	if err != nil {
+		log.Error("Failed to revive trie", "err", err)
+	}
+	t.root = newRoot
+}
+
+func (t *Trie) revive(n node, key []byte, prefixKeyHex []byte, proofList [][]byte, pos int) (node, bool, error) {
+
+	if pos > len(prefixKeyHex) {
+		return nil, false, fmt.Errorf("target revive node not found")
+	}
+
+	if pos == len(prefixKeyHex) {
+		hn, ok := n.(hashNode)
+		if !ok {
+			return nil, false, fmt.Errorf("prefix key path does not lead to a hash node")
+		}
+
+		revivedNode, err := VerifyPathProof(common.BytesToHash(hn), key, prefixKeyHex, proofList)
```

Comment:
Why not verify first then revive? If there is any error in `proofList`, just return.